src/FluentNHibernate/Automapping/AutoMapper.cs

```diff
@@ -78,8 +78,7 @@ private void MapInheritanceTree(Type classType, ClassMappingBase mapping, IList<
                 }
 
                 SubclassMapping subclassMapping;
-                var tempSubClassMap = mapping as SubclassMapping;
-                if(!tempIsNull && tempMapping.IsUnionSubclass || tempSubClassMap != null && tempSubClassMap.SubclassType == SubclassType.UnionSubclass)
+                if(!tempIsNull && tempMapping.IsUnionSubclass || mapping is SubclassMapping tempSubClassMap&& tempSubClassMap.SubclassType == SubclassType.UnionSubclass)

```

Comment:
@pawan-deepsource this is failing with:

```
C:\projects\fluent-nhibernate\src\FluentNHibernate\Automapping\AutoMapper.cs(86,45): error CS0165: Use of unassigned local variable 'tempSubClassMap' [C:\projects\fluent-nhibernate\src\FluentNHibernate\FluentNHibernate.csproj]
C:\projects\fluent-nhibernate\src\FluentNHibernate\Automapping\AutoMapper.cs(86,45): error CS0165: Use of unassigned local variable 'tempSubClassMap' [C:\projects\fluent-nhibernate\src\FluentNHibernate\FluentNHibernate.csproj]
C:\projects\fluent-nhibernate\src\FluentNHibernate\Automapping\AutoMapper.cs(86,45): error CS0165: Use of unassigned local variable 'tempSubClassMap' [C:\projects\fluent-nhibernate\src\FluentNHibernate\FluentNHibernate.csproj]
```

Reply:
(in a next condition)

Reply:
This case is actually a false positive since there's a reference to `tempSubClassMap` after the cast. Autofix wouldn't have failed if the check hadn't fired in the first place. IMO, it was the check that mislead Autofix and that's where the problem is. Tracking it internally.